ui/admin/app/components/form/policy/policy-selection/index.js

```diff
@@ -13,41 +13,34 @@ export default class FormPolicySelectionComponent extends Component {
    * @type {boolean}
    */
   get showCustomInput() {
-    if (this.args.name === 'retention_policy') {
-      return this.isCustomRetentionSelected;
-    } else if (
-      this.args.name === 'deletion_policy' &&
-      this.args.model.retain_for?.days >= 0
-    ) {
-      return this.isCustomDeletionSelected;
-    } else {
-      return null;
-    }
+    return this.args.selectedOption === 'custom' ? true : false;
   }
 
   /**
-   * Returns true if rentention days are greater than 0
+   * Disable toggle when the form is disabled and when the retain/delete after days are 0
    * @type {boolean}
    */
-  get isCustomRetentionSelected() {
-    //no need to show custom input for the below days
-    const arr = [-1, 0, 2555, 2190];
-    return arr.includes(this.args.model.retain_for?.days) ? false : true;
+  get toggleDisabled() {
+    if (
+      this.args.disabled ||
+      this.args.model[this.args.customInputName]?.days === 0 ||
+      !this.args.model[this.args.customInputName]?.days
+    ) {
+      return true;
+    } else {
+      return false;
+    }
   }
 
-  /**
-   * Returns true if deletion days are greater than 0
-   * @type {boolean}
-   */
-  get isCustomDeletionSelected() {
-    return this.args.model.delete_after?.days > 0;
-  }
   /**
    * Returns true if the toggle is on
    * @type {boolean}
    */
   get isOverridable() {
-    return this.args.model[this.args.customInputName]?.overridable;
+    return (
+      this.args.model[this.args.customInputName]?.days &&
+      this.args.model[this.args.customInputName]?.overridable
+    );
   }
 
   get isDeleteDisable() {
```

Comment:
This could have also been simplified to:
```
    return this.args.name === 'deletion_policy' &&
          this.args.model.retain_for?.days === -1;
```